### ISSUE TITLE
Error mask split in subsampling

### ISSUE THREAD

**Issue by day9011** [espnet/espnet#3092]
in pytorch, get subsampling mask by split like: mask[:, :, :-2:2][:, :, :-2:2] where error after conv.
example:
lens = torch.LongTensor([[24], [40], [60], [100]]).cpu()
lens after conv:
tensor([[ 5],
        [ 9],
        [14],
        [24]])
mask = make_mask_by_length(a, lens).unsqueeze(-2)
new_mask = mask[:, :, :-2:2][:, :, :-2:2]
print(torch.sum(new_mask.int(), -1))
tensor([[ 6],
        [10],
        [15],
        [24]])

**Comment by sw005320** [comment]
Is that the same issue as https://github.com/espnet/espnet/issues/3092?
If so, we'll definitely revisit it.
